### PR TITLE
Cleanup of existing tests

### PR DESCRIPTION
**Pull request by bjosv**
* Added build target to start and stop test clusters needed for some tests
  `make start` - Start Redis clusters, one cluster in each Docker container
  `make stop` - Stop Redis clusters

* Rename async test and remove redundant example binaries
  Some examples was already covered by tests

* Change IPv6 example into a test

* Use common tests assert macros in tests

### CODE REVIEW

File: tests/ct_pipeline.c [hunk 1]
```diff
@@ -1,56 +1,51 @@
...
+#include "test_utils.h"
 
 #include <assert.h>
 #include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
-#define CLUSTER_NODE "127.0.0.1:30001"
+#define CLUSTER_NODE "127.0.0.1:7000"
```
Comment (zuiderkwast) [inline tests/ct_pipeline.c:11]: Why change the ports? Just curious.

The references to these ports in the README needs to be updated.

**Review by zuiderkwast — COMMENTED** [review]
Looks good!

It's a bit hard to see what is renamed files and what is new, but that's fine.

Please update the README with the new ports and the flags necessary to run all the tests, after rebasing.

File: README.md [hunk 2]
```diff
@@ -51,6 +51,15 @@ $ cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DENABLE_SSL=ON ..
 $ make
 ```
 
+Some tests needs a Redis cluster and that can be setup by the build targets `start`/`stop`
```
[the remark this answers is not in the document]
Comment (bjosv) [reply README.md:54]: good, changed

File: tests/ct_pipeline.c (hunk 1 above)
Comment (bjosv) [reply tests/ct_pipeline.c:11]: Updated readme.
A lot of examples and the redis docs uses 7000 as a base-port, like https://redis.io/topics/cluster-tutorial
Found that out when looking at https://github.com/Grokzen/docker-redis-cluster, which also uses 7000 as default

**Pull request merged**